internal/core/items/text.rs

```diff
@@ -326,13 +326,21 @@ impl Item for TextInput {
             return InputEventResult::EventIgnored;
         }
         match event {
-            MouseEvent::Pressed { position, button: PointerEventButton::Left } => {
+            MouseEvent::Pressed { position, button: PointerEventButton::Left, click_count } => {
                 let clicked_offset =
                     window_adapter.renderer().text_input_byte_offset_for_position(self, position)
                         as i32;
                 self.as_ref().pressed.set(true);
                 self.as_ref().anchor_position_byte_offset.set(clicked_offset);
-                self.set_cursor_position(clicked_offset, true, window_adapter, self_rc);
+
+                println!("click count {}", click_count);
+                match click_count {
+                    1 => self.set_cursor_position(clicked_offset, true, window_adapter, self_rc),
+                    2 => self.select_word(window_adapter, self_rc),
+                    3 => self.select_paragraph(window_adapter, self_rc),
+                    _ => {}
```

Comment:
I would say it  should be

```suggestion
                match click_count % 3 {
                    0 => self.set_cursor_position(clicked_offset, true, window_adapter, self_rc),
                    1 => self.select_word(window_adapter, self_rc),
                    2 => self.select_paragraph(window_adapter, self_rc),
                    _ => unreachable!()
```

Also, we need to remember the selection mode so that when we get further MouseEvent::Moved, we select words or lines.

Reply:
> Also, we need to remember the selection mode so that when we get further MouseEvent::Moved, we select words or lines.

@ogoffart should be do is in this PR?

Reply:
Can be done later